assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -658,57 +675,81 @@ const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
 
   x += spaceBetween + SLOT_WIDTH;
 
-  // Get the size of the diagram svg, excluding emphasis
-  let dimensions = document.getElementById("line-map")?.getBoundingClientRect();
-  let height = dimensions?.height ?? 0;
-  let width = dimensions?.width ?? 0;
+  // Get the size of the diagram line map svg, excluding emphasis
+  const [lineDiagramHeight, setLineDiagramHeight] = useState(0);
+  const [lineDiagramWidth, setLineDiagramWidth] = useState(0);
+
+  // Measures line-map svg when the scaleFactor changes, updates state
+  const measuredRef = useCallback(node => {
+    const myObserver = new ResizeObserver(() => {})
+
+    if (node !== null) {
+      myObserver.observe(node)
+      setLineDiagramHeight(node.getBoundingClientRect().height);
+      setLineDiagramWidth(node.getBoundingClientRect().width);
+    }
+    
+    return () => myObserver.disconnect()
+  }, [scaleFactor]);
```

Comment:
Ooooo... gosh, I see. I'm not sure if step 5 happens when we need it to, and I'm not sure how to fix that. Trying to figure out alternatives

Reply:
So far it's behaving deterministically, so I want to say that yes, step 5 is happening after the node is updated. But I don't have proof 😩

Reply:
Shoot, I may have just spotted a nondeterministic sizing. A refresh resolved it, so I do think it was this bug. Grrrr

Reply:
I think @cmaddox5 found the [missing info](https://elfi-y.medium.com/react-callback-refs-a-4bd2da317269) we need to feel more confident in this function. He quoted:

```
When using ref callback, instead of accessing the .current value directly, we now can manage the reference storage whenever we want. As per the documentation states: React will call the ref callback with the DOM element when the component mounts, and call it with null when it unmounts. Therefore, Refs are guaranteed to be up-to-date before componentDidMount or componentDidUpdate fires.
```